agent/command/git.go

```diff
@@ -579,6 +579,7 @@ func (c *gitFetchProject) fetchSource(ctx context.Context,
 	if err != nil {
 		return err
 	}
+	gitCommands = append([]string{fmt.Sprintf("chmod +w %s", c.Directory)}, gitCommands...)
```

Comment:
Also, this needs to `chmod -R` because `rm -rf` can fail due to permissions if any file inside the directory doesn't have write permissions.